pkg/shares/parse_compact_shares.go

```diff
@@ -3,20 +3,31 @@ package shares
 import (
 	"encoding/binary"
 	"errors"
+	"fmt"
 
 	"github.com/celestiaorg/celestia-app/pkg/appconsts"
 )
 
-// parseCompactShares takes raw shares and extracts out transactions,
+// parseCompactShares takes rawShares and extracts out transactions,
 // intermediate state roots, or evidence. The returned [][]byte do not have
 // namespaces, info bytes, data length delimiter, or unit length
 // delimiters and are ready to be unmarshalled
-func parseCompactShares(shares [][]byte) (data [][]byte, err error) {
-	if len(shares) == 0 {
+func parseCompactShares(rawShares [][]byte) (data [][]byte, err error) {
```

Comment:
[Non Blocking][Question]
I like the idea of checking the share version when parsing, since inside `ProcessProposal`, we don't have this information. However, since this might be used in multiple contexts, doesn't it make more sense to pass the supported share versions as a parameter? While providing a default one.
This way, we can parse shares for any version.

Reply:
good point. I made `supportedShareVersions` a param in https://github.com/celestiaorg/celestia-app/pull/837/commits/ff75095f6eef1d4f7e4f3c16168801e141ce6ecd

I only lifted this as far as the public `Parse*` methods because those are public and may be called from external consumers. It doesn't seem necessary for an external consumer (i.e. celestia-node) to pass in a different list of supportedShareVersions because that list is defined in this repo. In the future, when we have share version `1`, logic in this repo's `share` package will change to conditionally parse different share versions (i.e. `1` and `0` differently) and at the same time, we can add `1` to https://github.com/rootulp/celestia-app/blob/ff75095f6eef1d4f7e4f3c16168801e141ce6ecd/pkg/appconsts/appconsts.go#L140